server/src/crm/crm.service.ts

```diff
@@ -3,21 +3,39 @@ import {
   HttpStatus,
   Injectable
 } from '@nestjs/common';
-import * as zlib from 'zlib';
-import * as Request from 'request';
 import { ConfigService } from '../config/config.service';
 import { ADAL } from '../_utils/adal';
-import { XmlService } from './xml/xml.service';
+import { ORequest } from 'odata';
+import * as superagent from 'superagent';
+import { dateParser } from './crm.utilities';
+
+// custom parser for json data. instantiates javascript date objects for date-like json strings
+// code cribbed from https://github.com/visionmedia/superagent/issues/986#issuecomment-218435902
+superagent.parse['application/json'] = function parseJSON(res, fn) {
+  res.text = '';
+  res.setEncoding('utf8');
+  res.on('data', function (chunk) { res.text += chunk; });
+  res.on('end', function () {
+    let body, err;
+
+    try {
+      body = res.text && JSON.parse(res.text, dateParser);
+    } catch (e) {
+      err = e;
+      // return the raw response if the response parsing fails
+      err.rawResponse = res.text || null;
+
+      // return the http status code if the response parsing fails
+      err.statusCode = res.status;
+    } finally {
+      fn(err, body);
+    }
+  });
+};
```

Comment:
This code extends superagent's default JSON parser and applies the "dateParser" code to it. Note that this is the same behavior in the old, static crm-web-api code. Previously, this was called "_dateReviver". You can find this old method in the diff.